rfcs/0050-vm-syscalls-3/0050-vm-syscalls-3.md

```diff
@@ -12,174 +12,139 @@ Created: 2023-04-17
+
+By providing `NULL` as `fd` and a `size_t` pointer with a value of 0 as `count`, this syscall can be used to fetch the count of `inherited_fds` in the Spawn syscall. However, by providing `NULL` as `fd` and a non-zero value as `count`, this syscall fails.
 
-### Current Memory
-[Current Memory]: #current-memory
+### Read
+[Read]: #read
 
-Get the Current Memory Usage. The result is the sum of the memory usage of the parent script and the child script.
+This syscall reads data from a pipe via a file descriptor. The syscall Read attempts to read up to value pointed by length bytes from file descriptor fd into the buffer, and the actual length of data read is written back to the length parameter. The syscall may pause the execution of current process.
 
 ```c
-int ckb_current_memory();
+int ckb_read(uint64_t fd, void* buffer, size_t* length);
 ```
 
-The system call returns an integer. Multiply it by 0.5 to get the actual memory usage (unit: megabytes)
+### Write
+[Write]: #write
 
-### Set Content
-[Set Content]: #set-content
-
-The child script can return bytes data to the parent script through `Set Content`.
+This syscall writes data to a pipe via a file descriptor. The syscall Write writes up to value pointed by length bytes from the buffer, and the actual length of data written is written back to the length parameter. The syscall may pause the execution of current process.
 
 ```c
-int ckb_set_content(uint8_t* content, uint64_t* length);
+int ckb_write(uint64_t fd, const void* buffer, size_t* length);
 ```
 
-- Length up to 256K.
-- If the written length is greater than the limit given by *Spawn*, the final written length is the minimum of the two.
-- Returns 0 for success, otherwise for failure.
-- This function is optional. Not every child script needs to call this, and nothing will happen if you call it from the prime script.
+### Close
+[Close]: #close
 
-### Spawn example
+This syscall manually closes a file descriptor. After calling this, any attempt to read/write the file descriptor pointed to the other end would fail.
 
-Consider the creation of a dependency library with a straightforward function that receives parameters, concatenates them, and subsequently returns the resulting string to the caller.
+```c
+int ckb_close(uint64_t fd);
+```
 
-**lib_strcat.c**
+It's not always necessary to manually close file descriptors. When a process is terminated, all file descriptors owned by the process are automatically closed.
 
-```c
-#include <stdint.h>
-#include <string.h>
+### Wait
+[Wait]: #wait
 
-#include "ckb_syscalls.h"
+The syscall pauses until the execution of a process specified by `pid` has ended.
```

Comment:
What if parent A spawns child B, but child B terminates before parent A calls `ckb_wait`? I know you do have an answer, but I believe we should properly define the behavior here.